youtube_dl/extractor/vidello.py

```diff
@@ -0,0 +1,44 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+from ..utils import (
+    clean_html
+)
+
+
+class VidelloIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?embed.vidello\.com/[0-9]/(?P<id>[a-zA-Z0-9]+)/player.html'
+    _TEST = {
+        'url': 'https://embed.vidello.com/2/t1umm637xb1ylgw4/player.html',
+        'md5': '7a4d76ac74ef7724af4c6c3ecb5e0042',
+        'info_dict': {
+            'id': 't1umm637xb1ylgw4',
+            'ext': 'mp4',
+            'title': 'Vidello Hosting & Marketing',
+            'description': "Start marketing your videos more effectively on \x03the web utilising vidello's premium hosting, streaming, \x03analytics & marketing features to grow your \x03online business fast."
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        vidello_settings = self._parse_json(self._search_regex(
+            r'var\s*.+' + video_id + '[^=]+=\s*({.+});', webpage, 'vidello settings'), video_id)
```

Comment:
No. What's the point of video id here?

Reply:
but I thought this needs to be robust if they change the javascript. if they change the name to video_id_options it won't work anymore. But for now I will change it to settings\s*=\s*({.+}); as u suggested

Reply:
Video id is much more likely to disappear than `settings` to be renamed.